### PR TITLE
Gp 552 vendored deps usr local

### PR DESCRIPTION
**Pull request by bmdoil**
No longer required to build/install xerces-c when using
gpdb6-build container images.

xerces-c, zstd are installed in /usr/local/lib64

install_python is smarter about finding the vendored python in /opt

### CODE REVIEW

File: ci/concourse/scripts/compile-gpdb-oss.bash [hunk 1]
```diff
@@ -83,71 +68,38 @@ git_info() {
...
-	echo "Copying python from /opt/python-2.7.12 into ${greenplum_install_dir}/ext/python..."
-	cp --archive /opt/python-2.7.12/* "${greenplum_install_dir}/ext/python"
+	# Vendor shared libraries
+	for path in ${linker_search_path//:/ }; do if [[ -d $path ]] ; then for lib in "${VENDORED_LIBS[@]}"; do find $path -name $lib -exec cp -an '{}' ${greenplum_install_dir}/lib \;; done; fi; done
+	# Vendor headers
+	for header in "${VENDORED_HEADERS[@]}"; do find /usr/local/include/ /usr/include/ -name $header -exec cp -an '{}' ${greenplum_install_dir}/include \;; done
 
+	# Vendor python
+	echo "Copying python from ${PYTHONHOME} into ${greenplum_install_dir}/ext/python..."
+	cp --archive ${PYTHONHOME}/* "${greenplum_install_dir}/ext/python"
```
Comment (BaiShaoqi) [inline ci/concourse/scripts/compile-gpdb-oss.bash:97]: Just pick this line to check if there are any difference in the old pipeline and the dev pipeline , it shows ` /opt/python-2.7.12/share ` is not copied. 
in the old pipeline: https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/greenplum-database-release/jobs/compile_gpdb_centos7/builds/43
```
cp --archive /opt/python-2.7.12/bin /opt/python-2.7.12/include /opt/python-2.7.12/lib /opt/python-2.7.12/share /usr/local/greenplum-db-6.16.2/ext/python

```
in the dev pipeline: https://releng.ci.gpdb.pivotal.io/teams/main/pipelines/greenplum-database-release-GP-552-vendored-deps-usr-local-bdoil/jobs/compile_gpdb_centos7/builds/15**
```
cp --archive /opt/python-2.7.12/bin /opt/python-2.7.12/include /opt/python-2.7.12/lib /usr/local/greenplum-db-6.16.2+dev.44.g60f31c0/ext/python
```

**Review by BaiShaoqi — COMMENTED** [review]
Overall the code change looks good to me, suggest to check if there is difference in the final gpdb tarball.

File: ci/concourse/scripts/compile-gpdb-oss.bash (hunk 1 above)
Comment (bmdoil) [reply ci/concourse/scripts/compile-gpdb-oss.bash:108]: Fixed by updating the python source artifact

**Review by bhanu-atturu — APPROVED** [review]
+1

**Pull request merged**